yt/frontends/gadget/data_structures.py

```diff
@@ -409,20 +402,14 @@ def _parse_parameter_file(self):
                 omega_lambda=self.omega_lambda,
             )
             self.current_time = cosmo.lookback_time(self.current_redshift, 1e6)
-            only_on_root(
-                mylog.info,
+            mylog.info(
                 "Calculating time from %0.3e to be %0.3e seconds",
                 hvals["Time"],
                 self.current_time,
             )
         self.parameters = hvals
 
-        prefix = os.path.abspath(
-            os.path.join(
-                os.path.dirname(self.parameter_filename),
-                os.path.basename(self.parameter_filename).split(".", 1)[0],
-            )
-        )
+        prefix = os.path.join(self.directory, self.basename.split(".", 1)[0])
```

Comment:
A pet-peeve of mine. Not needed if it degrades readability to others.
```suggestion
        basename, _, _ = self.basename.partition(".")
        prefix = os.path.join(self.directory, basename)
```